pkg/controller/enterprisesearch/enterprisesearch_controller_test.go

```diff
@@ -280,6 +282,57 @@ func TestReconcileEnterpriseSearch_Reconcile_Create_Update_Resources(t *testing.
 	checkResources()
 }
 
+func TestReconcileEnterpriseSearch_doReconcile_AssociationDelaysVersionUpgrade(t *testing.T) {
```

Comment:
Enterprise Search is the only driver with great unit test coverage, so I also added a test there.
Other controllers don't really have unit tests for their main reconciliation algorithm, but E2E tests do cover it in a different way.